packages/flutter/lib/src/integrations/replay_log_integration.dart

```diff
@@ -0,0 +1,63 @@
+// ignore_for_file: invalid_use_of_internal_member
+
+import 'package:sentry/sentry.dart';
+import '../sentry_flutter_options.dart';
+import '../native/sentry_native_binding.dart';
+
+/// Integration that adds replay-related information to logs using lifecycle callbacks
+class ReplayLogIntegration implements Integration<SentryFlutterOptions> {
+  static const String integrationName = 'ReplayLog';
+
+  final SentryNativeBinding? _native;
+  ReplayLogIntegration(this._native);
+
+  SentryFlutterOptions? _options;
+  SdkLifecycleCallback<OnBeforeCaptureLog>? _addReplayInformation;
+
+  @override
+  Future<void> call(Hub hub, SentryFlutterOptions options) async {
+    if (!options.replay.isEnabled) {
+      return;
+    }
+    final sessionSampleRate = options.replay.sessionSampleRate ?? 0;
+    final onErrorSampleRate = options.replay.onErrorSampleRate ?? 0;
+
+    _options = options;
+    _addReplayInformation = (OnBeforeCaptureLog event) {
+      final scopeReplayId = hub.scope.replayId;
+      final replayId = scopeReplayId ?? _native?.replayId;
+      final replayIsBuffering = replayId != null && scopeReplayId == null;
+
+      if (sessionSampleRate > 0 && replayId != null && !replayIsBuffering) {
+        event.log.attributes['sentry.replay_id'] = SentryLogAttribute.string(
+          scopeReplayId.toString(),
+        );
+      } else if (onErrorSampleRate > 0 &&
+          replayId != null &&
+          replayIsBuffering) {
+        event.log.attributes['sentry.replay_id'] = SentryLogAttribute.string(
+          replayId.toString(),
+        );
+        event.log.attributes['sentry._internal.replay_is_buffering'] =
+            SentryLogAttribute.bool(true);
+      }
```

Comment:
should we add a little note that `replayIsBuffering` will always result to false for Cocoa until we get the chance to update the Cocoa SDK?